plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,302 @@
+			continue;
+		}
+
+		// If not passed, decide whether to try conditional request.
+		if ( false === $check ) {
+			// Only if no far-future headers at all, we try conditional request.
+			$conditional_pass = perflab_ffh_try_conditional_request( $asset, $headers );
+			if ( ! $conditional_pass ) {
+				$final_status   = 'recommended';
+				$fail_details[] = array(
+					'filename' => $filename,
+					'reason'   => __( 'No far-future headers and no conditional caching', 'performance-lab' ),
+				);
+			} else {
+				$final_status   = 'recommended';
+				$fail_details[] = array(
+					'filename' => $filename,
+					'reason'   => __( 'No far-future headers but conditionally cached', 'performance-lab' ),
+				);
+			}
+		} else {
+			// If there's a max-age or expires but below threshold, we skip conditional.
+			$final_status   = 'recommended';
+			$fail_details[] = array(
+				'filename' => $filename,
+				'reason'   => $check['reason'],
+			);
+		}
+	}
+
+	return array(
+		'final_status' => $final_status,
+		'details'      => $fail_details,
+	);
+}
+
+/**
+ * Checks if far-future expiration headers are enabled.
+ *
+ * @since n.e.x.t
+ *
+ * @param WpOrg\Requests\Utility\CaseInsensitiveDictionary|array<string, string|array<string>> $headers Response headers.
+ * @return array{passed: bool, reason: string}|false Detailed result. If passed=false, reason explains why it failed and false if no headers found.
+ */
+function perflab_ffh_check_headers( $headers ) {
+	/**
+	 * Filters the threshold for far-future headers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $threshold Threshold in seconds.
+	 */
+	$threshold = apply_filters( 'perflab_far_future_headers_threshold', YEAR_IN_SECONDS );
+
+	$cache_control = isset( $headers['cache-control'] ) ? $headers['cache-control'] : '';
+	$expires       = isset( $headers['expires'] ) ? $headers['expires'] : '';
+
+	// Check Cache-Control header for max-age.
+	$max_age = 0;
+	if ( '' !== $cache_control ) {
+		// There can be multiple cache-control headers, we only care about max-age.
+		$controls = is_array( $cache_control ) ? $cache_control : array( $cache_control );
+		foreach ( $controls as $control ) {
+			if ( (bool) preg_match( '/max-age\s*=\s*(\d+)/', $control, $matches ) ) {
+				$max_age = (int) $matches[1];
+				break;
+			}
+		}
+	}
+
+	// If max-age meets or exceeds the threshold, we consider it good.
+	if ( $max_age >= $threshold ) {
+		return array(
+			'passed' => true,
+			'reason' => '',
+		);
+	}
+
+	// If max-age is too low or not present, check Expires.
+	if ( is_string( $expires ) && '' !== $expires ) {
+		$expires_time = strtotime( $expires );
+		if ( (bool) $expires_time && ( $expires_time - time() ) >= $threshold ) {
+			// Good - Expires far in the future.
+			return array(
+				'passed' => true,
+				'reason' => '',
+			);
+		}
+
+		// Expires header exists but not far enough in the future.
+		if ( $max_age > 0 && $max_age < $threshold ) {
```

Comment:
This condition doesn't seem to be necessary:

```suggestion
		if ( $max_age > 0 ) {
```

Because above we're already doing:

```php
	if ( $max_age >= $threshold ) {
		return array(
			'passed' => true,
			'reason' => '',
		);
	}
```

So we already know that `$max_age` is less than the `$threshold`.